TASVideos.Common/Extensions/StringExtensions.cs

```diff
@@ -188,5 +188,17 @@ public static string ToSnakeCase(this string str)
 				.Concat(str.Select((s, i) => i > 0 && char.IsUpper(s) ? "_" + s : s.ToString()))
 				.ToLower();
 		}
+
+		public static string LastCommaToAmpersand(this string commaString)
+		{
+			int lastComma = commaString.LastIndexOf(",");
+
+			if (lastComma == -1)
+			{
+				return commaString;
+			}
+
+			return commaString[..lastComma] + " &" + commaString[(lastComma + 1)..];
```

Comment:
I know adelikat mentioned he can clean this up but this really should be a helper method elsewhere that does this in the same stroke as the join operation and probably using string remove and insert instead of the array ranges with inferred start/end

Reply:
Code suggestion (untested):
```csharp
public static string JoinListWithEndingAmpersand(this IEnumerable<string> list) {
	if (!list.TryGetNonEnumeratedCount(out var count)) return list.ToList().JoinListWithEndingAmpersand();
	switch (count) {
		case 1:
			return list.First();
		case 2:
			return $"{list.First()} {list.First()}";
		default:
			var head = list.Take(count - 1);
			return $"{string.Join(", ", head)}, and {list.First()}";
	}
}
```